transport.go

```diff
@@ -127,29 +143,25 @@ func envelopeFromBody(event *Event, dsn *Dsn, sentAt time.Time, body json.RawMes
 		return nil, err
 	}
 
-	var itemType string
-	switch event.Type {
-	case transactionType:
-		itemType = transactionType
-	default:
-		itemType = eventType
+	if event.Type == transactionType {
+		err = encodeEnvelopeItem(enc, transactionType, body)
+	} else {
+		err = encodeEnvelopeItem(enc, eventType, body)
 	}
-
-	// Item header
-	err = enc.Encode(struct {
-		Type   string `json:"type"`
-		Length int    `json:"length"`
-	}{
-		Type:   itemType,
-		Length: len(body),
-	})
 	if err != nil {
 		return nil, err
 	}
-	// payload
-	err = enc.Encode(body)
-	if err != nil {
-		return nil, err
+
+	// Profile data
+	if event.transactionProfile != nil {
+		body, err = json.Marshal(event.transactionProfile)
```

Comment:
I believe if you marshal before you pass it to `encodeEnvelopeItem`, it'll be sent as a string of JSON data, not as a JSON object. I don't think we need to marshal before we pass it to `encodeEnvelopeItem`.

Reply:
It won't encode twice, the body is passed as `json.RawMessage` which means it won't encoded again. It's been done the same way for actual event & transaction payload before. Also, it is done this way so that we can get the length of the serialized body and set it in the envelope item header.

Anyway, this reminded me there were no tests checking the transaction profile being part of the envelope so I've amended that.

Reply:
You can see the profile envelope item here: [`54d52ac` (#626)](https://github.com/getsentry/sentry-go/pull/626/commits/54d52acd6020be85aba5c2fa93128295e314f74b#diff-89ae70865ace384239355ee0c882acef79b281a7c88c53c22058d75941cd2643R240)